core/commands/pin/pin.go

```diff
@@ -57,13 +57,32 @@ var addPinCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
 		Tagline:          "Pin objects to local storage.",
 		ShortDescription: "Stores an IPFS object(s) from a given path locally to disk.",
+		LongDescription: `
+Create a pin for the given object, protecting resolved CID from being garbage
+collected.
+
+An optional name can be provided, and read back via 'ipfs pin ls --detailed'.
+
+Be mindful of defaults:
+- Default pin type is 'recursive' (entire DAG).
+  - Pass -r=false to create a direct pin for a single block.
+  - Use 'pin ls -t recursive' to only list roots of recursively pinned DAGs (faster)
+- Default pin name is empty. Pass '--name' to 'pin add' to set one
+  and use 'pin ls --detailed' to see it.
+  - Pin add is idempotent: pinning CID which is already pinned won't change
+    the name, value passed with '--name' with the original pin is preserved.
+    To rename pin, use 'pin rm' and 'pin add --name'.
```

Comment:
@hacdias I've added this because running `ipfs pin add --name A CID` and then `ipfs pin add --name B CID` does not update the name (it is still `A`).

If you feel we could get follow-up calls update the name withoiut huge refactor, that would be preferable, and we could remove this sentence.